From 90fa0fe50f53bcb6eb386c7a20dbf87c5940c156 Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <michal.hrusecky@turris.com>
Date: Wed, 6 Sep 2023 23:19:21 +0200
Subject: [PATCH] target/linux: Rebase currently broken patches

Following kernel patch cannot be applied and had to be rebased:

backport-5.15/792-03-v6.6-net-phylink-add-pcs_enable-pcs_disable-methods.patch

Following kernel patch is no longer needed:

backport-5.15/792-v5.16-net-dpaa2-mac-add-support-for-more-10G-modes.patch
---
 ...k-add-pcs_enable-pcs_disable-methods.patch | 30 +++++++++-------
 ...2-mac-add-support-for-more-10G-modes.patch | 34 -------------------
 2 files changed, 17 insertions(+), 47 deletions(-)
 delete mode 100644 target/linux/generic/backport-5.15/792-v5.16-net-dpaa2-mac-add-support-for-more-10G-modes.patch

diff --git a/target/linux/generic/backport-5.15/792-03-v6.6-net-phylink-add-pcs_enable-pcs_disable-methods.patch b/target/linux/generic/backport-5.15/792-03-v6.6-net-phylink-add-pcs_enable-pcs_disable-methods.patch
index ceec58466e..f69d144996 100644
--- a/target/linux/generic/backport-5.15/792-03-v6.6-net-phylink-add-pcs_enable-pcs_disable-methods.patch
+++ b/target/linux/generic/backport-5.15/792-03-v6.6-net-phylink-add-pcs_enable-pcs_disable-methods.patch
@@ -14,8 +14,10 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  include/linux/phylink.h   | 16 +++++++++++++
  2 files changed, 55 insertions(+), 9 deletions(-)
 
---- a/drivers/net/phy/phylink.c
-+++ b/drivers/net/phy/phylink.c
+Index: linux-5.15.130/drivers/net/phy/phylink.c
+===================================================================
+--- linux-5.15.130.orig/drivers/net/phy/phylink.c
++++ linux-5.15.130/drivers/net/phy/phylink.c
 @@ -34,6 +34,10 @@ enum {
  	PHYLINK_DISABLE_STOPPED,
  	PHYLINK_DISABLE_LINK,
@@ -35,7 +37,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  
  	bool mac_link_dropped;
  	bool using_mac_select_pcs;
-@@ -795,6 +800,22 @@ static void phylink_mac_pcs_an_restart(s
+@@ -806,6 +811,22 @@ static void phylink_mac_pcs_an_restart(s
  	}
  }
  
@@ -58,7 +60,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  static void phylink_major_config(struct phylink *pl, bool restart,
  				  const struct phylink_link_state *state)
  {
-@@ -832,12 +853,16 @@ static void phylink_major_config(struct
+@@ -843,12 +864,16 @@ static void phylink_major_config(struct
  	 * for the change.
  	 */
  	if (pcs_changed) {
@@ -75,7 +77,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  	if (pl->pcs_ops) {
  		err = pl->pcs_ops->pcs_config(pl->pcs, pl->cur_link_an_mode,
  					      state->interface,
-@@ -1260,6 +1285,7 @@ struct phylink *phylink_create(struct ph
+@@ -1277,6 +1302,7 @@ struct phylink *phylink_create(struct ph
  	pl->link_config.speed = SPEED_UNKNOWN;
  	pl->link_config.duplex = DUPLEX_UNKNOWN;
  	pl->link_config.an_enabled = true;
@@ -83,7 +85,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  	pl->mac_ops = mac_ops;
  	__set_bit(PHYLINK_DISABLE_STOPPED, &pl->phylink_disable_state);
  	timer_setup(&pl->link_poll, phylink_fixed_poll, 0);
-@@ -1651,6 +1677,8 @@ void phylink_start(struct phylink *pl)
+@@ -1668,6 +1694,8 @@ void phylink_start(struct phylink *pl)
  	if (pl->netdev)
  		netif_carrier_off(pl->netdev);
  
@@ -92,16 +94,16 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  	/* Apply the link configuration to the MAC when starting. This allows
  	 * a fixed-link to start with the correct parameters, and also
  	 * ensures that we set the appropriate advertisement for Serdes links.
-@@ -1661,6 +1689,8 @@ void phylink_start(struct phylink *pl)
+@@ -1678,6 +1706,8 @@ void phylink_start(struct phylink *pl)
  	 */
  	phylink_mac_initial_config(pl, true);
  
 +	pl->pcs_state = PCS_STATE_STARTED;
 +
- 	clear_bit(PHYLINK_DISABLE_STOPPED, &pl->phylink_disable_state);
- 	phylink_run_resolve(pl);
+ 	phylink_enable_and_run_resolve(pl, PHYLINK_DISABLE_STOPPED);
  
-@@ -1680,16 +1710,9 @@ void phylink_start(struct phylink *pl)
+ 	if (pl->cfg_link_an_mode == MLO_AN_FIXED && pl->link_gpio) {
+@@ -1696,16 +1726,9 @@ void phylink_start(struct phylink *pl)
  			poll = true;
  	}
  
@@ -120,7 +122,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  	if (poll)
  		mod_timer(&pl->link_poll, jiffies + HZ);
  	if (pl->phydev)
-@@ -1726,6 +1749,10 @@ void phylink_stop(struct phylink *pl)
+@@ -1742,6 +1765,10 @@ void phylink_stop(struct phylink *pl)
  	}
  
  	phylink_run_resolve_and_disable(pl, PHYLINK_DISABLE_STOPPED);
@@ -131,8 +133,10 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  }
  EXPORT_SYMBOL_GPL(phylink_stop);
  
---- a/include/linux/phylink.h
-+++ b/include/linux/phylink.h
+Index: linux-5.15.130/include/linux/phylink.h
+===================================================================
+--- linux-5.15.130.orig/include/linux/phylink.h
++++ linux-5.15.130/include/linux/phylink.h
 @@ -419,6 +419,8 @@ struct phylink_pcs {
  /**
   * struct phylink_pcs_ops - MAC PCS operations structure.
diff --git a/target/linux/generic/backport-5.15/792-v5.16-net-dpaa2-mac-add-support-for-more-10G-modes.patch b/target/linux/generic/backport-5.15/792-v5.16-net-dpaa2-mac-add-support-for-more-10G-modes.patch
deleted file mode 100644
index 0bd96f1f4f..0000000000
--- a/target/linux/generic/backport-5.15/792-v5.16-net-dpaa2-mac-add-support-for-more-10G-modes.patch
+++ /dev/null
@@ -1,34 +0,0 @@
-From c314138bd045e050432158ab021160de3ba51c5e Mon Sep 17 00:00:00 2001
-From: Russell King <rmk+kernel@armlinux.org.uk>
-Date: Thu, 30 Jan 2020 22:42:38 +0000
-Subject: [PATCH 2/4] net: dpaa2-mac: add support for more 10G modes
-
-Phylink documentation says:
- * Note that the PHY may be able to transform from one connection
- * technology to another, so, eg, don't clear 1000BaseX just
- * because the MAC is unable to BaseX mode. This is more about
- * clearing unsupported speeds and duplex settings. The port modes
- * should not be cleared; phylink_set_port_modes() will help with this.
-
-So add the missing 10G modes.
-
-Signed-off-by: Russell King <rmk+kernel@armlinux.org.uk>
----
- drivers/net/ethernet/freescale/dpaa2/dpaa2-mac.c | 6 ++++++
- 1 file changed, 6 insertions(+)
-
---- a/drivers/net/ethernet/freescale/dpaa2/dpaa2-mac.c
-+++ b/drivers/net/ethernet/freescale/dpaa2/dpaa2-mac.c
-@@ -140,6 +140,12 @@ static void dpaa2_mac_validate(struct ph
- 	case PHY_INTERFACE_MODE_10GBASER:
- 	case PHY_INTERFACE_MODE_USXGMII:
- 		phylink_set(mask, 10000baseT_Full);
-+		phylink_set(mask, 10000baseKR_Full);
-+		phylink_set(mask, 10000baseCR_Full);
-+		phylink_set(mask, 10000baseSR_Full);
-+		phylink_set(mask, 10000baseLR_Full);
-+		phylink_set(mask, 10000baseLRM_Full);
-+		phylink_set(mask, 10000baseER_Full);
- 		if (state->interface == PHY_INTERFACE_MODE_10GBASER)
- 			break;
- 		phylink_set(mask, 5000baseT_Full);
-- 
2.42.0

